yaml-mode-expansions.el

```diff
@@ -0,0 +1,156 @@
+;;; yaml-mode-expansions.el --- expansions for yaml mode
+
+;; Copyright (C) 2021 Aaron Gonzales
+
+;; Author: Aaron Gonzales
+;; Keywords: marking region yaml YAML expand
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;;  - Additions implemented here:
+;;    - er/mark-yaml-key-value
+;;    - er/mark-yaml-list-item
+;;    - er/mark-yaml-block
+
+;;; Code:
+
+(require 'expand-region-core)
+
+(defconst yaml-indent 2)
+
+(if (not (fboundp 'yaml-indent-offset))
+    (defalias 'yaml-indent-offset 'yaml-indent))
+
+(defvar er--yaml-key-value-regex
+  (rx (one-or-more
+       (any "0-9A-Za-z"))
+      ":"
+      (zero-or-more " ")
+      (one-or-more
+       (any "0-9A-Za-z" " '_-"))))
+
+(defvar er--yaml-list-item-regex
+  (rx (seq "- "
+           (one-or-more
+            (any "0-9A-Za-z" " '_-"))
+           "\n")))
+
+(defvar er--yaml-block-regex
+  (rx (seq (zero-or-more
+            (any " -"))
+           (one-or-more
+            (any "0-9A-Za-z" " '_-"))
+           ":\n")))
+
+(defun er--get-regex-indentation-level (regex)
+  "Return the indentation level of the code with respect to the REGEX passed."
+  (if (looking-at regex)
+      ;; Block start means that the next level is deeper.
+      (+ (current-indentation) yaml-indent-offset)
+    ;; Assuming we're inside the block that we want to mark
+    (current-indentation)))
+
+(defun er/mark-yaml-line-base (regex)
+  "Mark line of yaml file based on simple REGEX."
+  (back-to-indentation)
+  (if (looking-at regex)
```

Comment:
`when` is more idiomatic here

Reply:
change added! Just curious `when` is more preferred when there is no `else` portion to eval right?